src/mat/4C_mat_micromaterialgp_static.cpp

```diff
@@ -354,12 +410,27 @@ void Mat::MicroMaterialGP::write_restart()
   std::shared_ptr<MultiScale::MicroStatic> microstatic =
       global_micro_state().microstaticmap_[microdisnum_];
 
-  // set displacements and EAS data of last step
+  // set displacements and history data of last step
   microstatic->set_state(dis_, disn_, stress_data_node_postprocessed_,
       stress_data_element_postprocessed_, strain_data_node_postprocessed_,
       strain_data_element_postprocessed_, plstrain_, nullptr);
   microstatic->write_restart(micro_output_, time_, step_, dt_);
 
+  // add element history data to restart output
+  if (!history_data_->empty())
+  {
+    std::shared_ptr<Core::FE::Discretization> discret =
+        (Global::Problem::instance(microdisnum_))->get_dis("structure");
+
+    Core::Communication::PackBuffer data;
+    for (Core::Elements::Element* actele : discret->my_row_element_range())
+    {
+      add_to_pack(data, (*(*history_data_)[actele->id()]));
+    }
```

Comment:
with my suggestion, this reduces to

```suggestion
    add_to_pack(data, history_data_);
```

Reply:
History data during runtime is in element column layout, whereas restart output is written in element row layout. This way will likely lead to trouble.